src/Microsoft.Kiota.Cli.Commons/IO/Console.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+
+namespace Microsoft.Kiota.Cli.Commons.IO;
+
+/// <summary>
+/// Console abstraction to help unit test.
+/// </summary>
+public interface IConsole {
+    /// <summary>
+    /// Writes the specified string value to the standard output stream.
+    /// </summary>
+    /// <param name="value">The value to write.</param>
+    void Write (string? value);
```

Comment:
```suggestion
    Task WriteAsync (string? value);
```
Does it make sense to make these methods async?

Reply:
I believe that would be the only one without the async version. Which you could simply return `Task.CompletedTask`. However, others would have the benefit of using the async version if you choose to.